src/IO/Progress.cpp

```diff
@@ -68,10 +68,12 @@ bool Progress::incrementPiecewiseAtomically(const Progress & rhs)
 {
     read_rows += rhs.read_rows;
     read_bytes += rhs.read_bytes;
-    read_raw_bytes += rhs.read_raw_bytes;
 
-    total_rows_to_read += rhs.total_rows_to_read;
-    total_raw_bytes_to_read += rhs.total_raw_bytes_to_read;
+    if (rhs.total_rows_to_read)
+        total_rows_to_read.store(rhs.total_rows_to_read);
+
+    if (rhs.total_bytes_to_read)
+        total_bytes_to_read.store(rhs.total_bytes_to_read);
```

Comment:
Is it because total_rows_to_read should be always the same (approximately) for progresses and it was a bug that we summed them?
And I think it's not good to do `if (x.load()) y.store(x.load())` because it can lead to race condition (x could be changed to 0 after `if` and before `store`). Most likely it's impossible in this place, but in general It's better to do smth like `if (auto value = x.load()) y.store(value)`

Reply:
And seems like tests failure are connected with this change (wrong total_rows_to_read value)